### DISCUSSION TITLE
Types for Custom Label props

### DISCUSSION THREAD

**Discussion by mfialko** [recharts/recharts#5223] [Q&A]
Good day!

I am trying to properly type the props for my custom label component.

Here is sandbox with simplified chart: [Link](https://codesandbox.io/p/sandbox/two-simple-pie-chart-forked-hh9fcg)

I've added `PieLabelRenderProps` and extended an interface with my custom props, but get typescript errors:

<img width="682" alt="Screenshot 2024-11-07 at 22 52 59" src="https://github.com/user-attachments/assets/d7a9a02f-286d-4b73-9c5e-ce54b98d3970">

<img width="643" alt="Screenshot 2024-11-07 at 22 52 38" src="https://github.com/user-attachments/assets/cfef0066-57cc-4823-80b0-4b52d9d00cc6">

`PieLabelRenderProps` defines `outerRadius` as `string | number`, allowing values like 50 or '70%'. However, in the component, this value is always received as a calculated number.

Is this an issue? How should this be handled?

**Comment by ckifer** [comment]
~~Hi, I don't see any error.~~

https://codesandbox.io/p/sandbox/two-simple-pie-chart-forked-hh9fcg

~~Can you upgrade the sandbox to the latest recharts version and then try?~~

Just kidding, codesandbox took a while to start the TS server

You can do something like this
```tsx
 label={(props) => <CustomizedLabel chartData={data01} {...props} />}
```

Looking at the definition here https://github.com/recharts/recharts/blob/master/src/polar/Pie.tsx#L62 its most likely because of that - it wants SVGProps rather than label props.

I would just do it the above way where you have more control. I'll take a look at the types though.

**Comment by ckifer** [comment]
For the second question it looks like we need to replace the current versions of those properties with this https://github.com/recharts/recharts/blob/3.x/src/polar/Pie.tsx#L195-L201
